core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -183,6 +183,17 @@ abstract class TaskContext extends Serializable {
   @Evolving
   def resources(): Map[String, ResourceInformation]
 
+  /**
+   * The shuffle generation ID of the stage that this task belongs to, it returns the stage
+   * attempt number while the stage is not determinate and returns none on the contrary.
+   */
+  def getShuffleGenerationId(shuffleId: Int): Option[Int] = {
+    val id = getLocalProperty(SparkContext.SHUFFLE_GENERATION_ID_PREFIX + shuffleId)
+    if (id != null) {
```

Comment:
nit: `Option(getLocalProperty(SparkContext.SHUFFLE_GENERATION_ID_PREFIX + shuffleId)).map(_.toInt)`

Reply:
Thanks, done in 5a4b8b8.